api/observability/v1/output_types.go

```diff
@@ -210,15 +210,48 @@ type BaseOutputTuningSpec struct {
 
 	// MinRetryDuration is the minimum time to wait between attempts to retry after delivery a failure.
 	//
+	// A duration string is a possibly signed sequence of decimal numbers, each with optional fraction
+	// and a unit suffix, such as "300ms", "-1.5h" or "2h45m". Valid time units are "ns", "us" (or
+	// "µs"), "ms", "s", "m", "h".
+	//
 	// +kubebuilder:validation:Optional
 	// +operator-sdk:csv:customresourcedefinitions:type=spec,displayName="Minimum Retry Duration"
 	MinRetryDuration *time.Duration `json:"minRetryDuration,omitempty"`
 
 	// MaxRetryDuration is the maximum time to wait between retry attempts after a delivery failure.
 	//
+	// A duration string is a possibly signed sequence of decimal numbers, each with optional fraction
+	// and a unit suffix, such as "300ms", "-1.5h" or "2h45m". Valid time units are "ns", "us" (or
+	// "µs"), "ms", "s", "m", "h".
+	//
 	// +kubebuilder:validation:Optional
 	// +operator-sdk:csv:customresourcedefinitions:type=spec,displayName="Maximum Retry Duration"
 	MaxRetryDuration *time.Duration `json:"maxRetryDuration,omitempty"`
+
+	// MaxBufferSize is the maximum size in bytes allowed for buffering unsent data to an output.
+	// It is written as a [Kubernetes quantity], for example: 128974848, 129e6, 129M,  123Mi
+	//
+	// NOTE: There are trade-offs to changing buffer sizes.
+	// The defaults are calculated to provide good performance in most cases.
+	// It is not recommended to set this field unless you have done tests
+	// to determine a value that performs better than the defaults in your specific setting.
+	// See also []
+	//
+	// The following are some trade-offs to bear in mind:
+	//
+	// - MaxBufferSize is per-output, add up all the outputs for the total impact per-node.
+	// - With `deliveryMode: AtLeastOnce`, buffers are created on the node's `/var/log` partition.
+	//   Filling that partition can have severe consequences for the entire node.
+	// - With `deliveryMode: AtLeastOnce` the original log file itself is used as a replay buffer
```

Comment:
This is not exactly correct since we were not able to impl acks given kubernetes_logs does not support acks at this time.  You could still lose logs collected but not sent

Reply:
Argh, I did not know that. I assumed it was supported because it is for file logs, and k8s are file logs with decorations. That is unfortunate. I'll remove the comment.